lab/dp/problems/custi/main.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Nicolae Mara-Ioana <maraioana9967@gmail.com>
+
+// Custi - Programare dinamica
+// Complexitate: O(n^2)
+
+// https://www.infoarena.ro/problema/custi
+
+#include <bits/stdc++.h>
+
+class Task {
+ public:
+    void solve() {
+        read_input();
+        get_result();
+        print_output();
+    }
+
+ private:
+    int n; // dimensiunea matricei
+    
+    std::vector<std::vector<int>> matrix; // matricea de input
+
+    // fiecare element solution[i] reprezinta numarul de custi
+    // de dimensiune maxima i x i ce se pot forma
+    // fiecare celula se foloseste o singura data ca si colt dreapta jos
+    // a unei custi, aceasta fiind cea de dimesniune maxima
+    std::vector<int> solution;
+
+    // citeste datele de intare din fisier si populeaza matricea
+    void read_input() { 
+        std::cin >> n;
+        matrix = std::vector<std::vector<int>>(n, std::vector<int>(n, 0));
+        solution = std::vector<int>(n + 1, 0);
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                std::cin >> matrix[i][j];
+            }
+        }
+    }
+
+    // ideea de la care am plecat: fiecare element este testat daca poate
+    // reprezenta coltul din dreapta jos al unui patrat, astfel:
+    // CAZUL DE BAZA:
+    // dp[i][0] = matrix[i][0]
+    // dp[0][i] = matrix[0][i]
+    // FORMULA: dp[i][j] = min(celelalte 3 colturi) + 1
+    //              (daca elementul poate defini un dreptunghi)
+    //          altfel dp[i][j] = 0
+    void get_result() {
+        // initializarea tabloului dp
+        std::vector<std::vector<int>> dp(n, std::vector<int>(n, 0)); 
+
+        if (matrix[0][0]) {
+            solution[1]++;
+        }
+
+        // tratarea cazului de baza: elementele de pe prima linie/coloana
+        // ce au valori nenule vor reprezenta custi de dimensiune 1 x 1,
+        // deoarece aceste elemente nu pot reprezenta 
+        // coltul din dreapta jos al unui patrat mai mare
+        dp[0][0] = matrix[0][0];
+        for (int i = 1; i < n; i++) {
+            if (matrix[i][0]) {
+                solution[1]++;
+            }
+            if (matrix[0][i]) {
+                solution[1]++;
+            }
+            dp[0][i] = matrix[0][i];
+            dp[i][0] = matrix[i][0];
+        }
+
+        // actualizarea tabloului conform formulei
+        // se parcurg toate elementele matricei
+        for (int i = 1; i < n; i++) {
+            for (int j = 1; j < n; j++) {
+                // elementul este valid, are valoarea 1,
+                // deci poate reprezenta coltul din dreapta jos
+                if (matrix[i][j]) {
+                    // se actualizeaza tabloul dp, cat si tabloul solutiei finale
+                    dp[i][j] = std::min(dp[i - 1][j - 1], 
+                        std::min(dp[i - 1][j], dp[i][j - 1])) + 1;
```

Comment:
Please run clang-format on your code! I think the entire formula can be on the same line.